lib/inky/inky.ex

```diff
@@ -0,0 +1,625 @@
+
+    # little endian, unsigned short
+    packed_height = [
+      :binary.encode_unsigned(Enum.fetch!(state.resolution_data, 1), :little),
+      <<0x00>>
+    ]
+
+    # Skipped map ord thing for packed_height..
+    # IO.puts("Starting to send shit..")
+
+    # Set analog block control
+    # IO.inspect("# Set analog block control")
+    send_command(state, 0x74, 0x54)
+
+    # Set digital block control
+    # IO.inspect("# Set digital block control")
+    send_command(state, 0x7E, 0x3B)
+
+    # Gate setting
+    # IO.inspect("# Gate setting")
+    send_command(state, 0x01, :binary.list_to_bin(packed_height ++ [0x00]))
+
+    # Gate driving voltage
+    # IO.inspect("# Gate driving voltage")
+    send_command(state, 0x03, [0b10000, 0b0001])
+
+    # Dummy line period
+    # IO.inspect("# Dummy line period")
+    send_command(state, 0x3A, 0x07)
+
+    # Gate line width
+    # IO.inspect("# Gate line width")
+    send_command(state, 0x3B, 0x04)
+
+    # Data entry mode setting 0x03 = X/Y increment
+    # IO.inspect("# Data entry mode setting 0x03 = X/Y increment")
+    send_command(state, 0x11, 0x03)
+
+    # Power on
+    # IO.inspect("# Power on")
+    send_command(state, 0x04)
+
+    # VCOM Register, 0x3c = -1.5v?
+    # IO.inspect("# VCOM Register, 0x3c = -1.5v?")
+    send_command(state, 0x2C, 0x3C)
+    send_command(state, 0x3C, 0x00)
+
+    # Always black border
+    # IO.inspect("# Always black border")
+    send_command(state, 0x3C, 0x00)
+
+    # Set voltage of VSH and VSL on Yellow device
+    if state.color == :yellow do
+      send_command(state, 0x04, 0x07)
+    end
+
+    # Set LUTs
+    # IO.inspect("# Set LUTs")
+    send_command(state, 0x32, get_luts(state.color))
+
+    # Set RAM X Start/End
+    # IO.inspect("# Set RAM X Start/End")
+    send_command(state, 0x44, :binary.list_to_bin([0x00, trunc(state.columns / 8) - 1]))
+
+    # Set RAM Y Start/End
+    # IO.inspect("# Set RAM Y Start/End")
+    send_command(state, 0x45, :binary.list_to_bin([0x00, 0x00] ++ packed_height))
+
+    # 0x24 == RAM B/W, 0x26 == RAM Red/Yellow/etc
+    for data <- [{0x24, buffer_a}, {0x26, buffer_b}] do
+      {cmd, buffer} = data
+
+      # Set RAM X Pointer start
+      # IO.inspect("# Set RAM X Pointer start")
+      send_command(state, 0x4E, 0x00)
+
+      # Set RAM Y Pointer start
+      # IO.inspect("# Set RAM Y Pointer start")
+      send_command(state, 0x4F, <<0x00, 0x00>>)
+      # IO.inspect("# Buffer thing")
+      send_command(state, cmd, buffer)
+    end
+
+    # Display Update Sequence
+    # IO.inspect("# Display Update Sequence")
+    send_command(state, 0x22, 0xC7)
+
+    # Trigger Display Update
+    # IO.inspect("# Trigger Display Update")
+    send_command(state, 0x20)
+
+    # Wait Before Deep Sleep
+    :timer.sleep(50)
+    busy_wait(state)
+
+    # Enter Deep Sleep
+    # IO.inspect("# Enter deep sleep")
+    send_command(state, 0x10, 0x01)
+  end
```

Comment:
If you wanted you could extract all these commands as separate functions, maybe in a separate module `Inky.Commands`. It would certainly remove the need for the comments explaining what this command does.

But I get if you are not in the mood to do that.

Reply:
The comments are from the original and have been my life-blood in understanding what is going on.

But in the vein of what you suggested for setup I could probably refactor to something like:
```
state
|> set_analog_block_control(0x74, 0x54)
|> set_digital_block_control(0x7E, 0x3B)
...
```

It would make the flow easier to follow I imagine. And all of those could possibly be move out to a module.